libusb/hid.c

```diff
@@ -1043,9 +1043,16 @@ hid_device * HID_API_EXPORT hid_open_path(const char *path)
 int HID_API_EXPORT hid_write(hid_device *dev, const unsigned char *data, size_t length)
 {
 	int res;
-	int report_number = data[0];
+	int report_number;
 	int skipped_report_id = 0;
 
+	if (!data || (length ==0)) {
+		errno = EINVAL;
```

Comment:
hidapi doen't use errno for error reporting, I would't introduce it randomly
Maybe introduce it once and for all functions properly, maybe as a separate PR

Reply:
I was afraid of that, not a problem.